pauga family reunion/pauga family reunion/Authentication/Sign In/SignInViewModel.swift

```diff
@@ -8,21 +8,80 @@
 import Foundation
 
 enum SignInLoadingState {
-    case loading, success, failure(Error)
+    case loading, finished
 }
 
 class SignInViewModel : ObservableObject {
     @Published var emailAddress = ""
     @Published var password = ""
-    @Published var loadingState: SignInLoadingState = .loading
+    @Published var loadingState: SignInLoadingState = .finished
+    @Published var showErrorAlert: Bool = false
+    @Published var errorMessage: String = ""
     
     func ctaTapped() {
         print("Sign In Button tapped")
-        test("tuli")
+        loadingState = .loading
+        guard validateEmail(),
+              validatePassword() else {
+            return
+        }
+        signUserIn()
     }
     
-    func test(_ name: String) {
-        print(name)
+    private let userService: UserServicable = UserService.shared
```

Comment:
Let's put all of our variables at the top of the file. See `SignUpViewModel` for an example of how we want to  organize our variables